### ISSUE TITLE
[Docs] modified Batched Conjugate Gradients (mBCG) in GPyTorch?

### ISSUE THREAD

**Issue by joacorapela** [cornellius-gp/gpytorch#941]
Hello,

I am trying to use GPyTorch to optimise code for Sparse 
Gaussian Process Factor Analysis (Duncker and Sahani, 2018). 
Specifically, I want to use the mBCG function described in
Gardner et al., 2018. As described in Equation 3 of this
paper, mBCG should take the input covariance (K_{XX}), the
system outputs (y) and a set of standard normal random
vectors (z_1, ..., z_t) as inputs and return the product of
the inverse of the input covariance with the outputs and the
random vectors.

My problem is that I cannot find an implementation of this 
function in the GPyTorch Github
https://github.com/cornellius-gp/gpytorch The closest
function I found is forward in class InvQuadLogDet. However,
this function returns the inverse quadratic form and not the
u_i or the T_i variables in Equation 3 in Gardner et al.,
2018.

I would greatly appreciate if you could tell me if/where can 
I find mBCG in GPyTorch?

Thanks, Joaquin

Lea Duncker and Maneesh Sahani. "Temporal alignment and 
latent Gaussian process factor inference in population spike
trains."  In Advances in Neural Information Processing
Systems (2018)

Gardner, Jacob R., Geoff Pleiss, David Bindel, Kilian Q. 
Weinberger, and Andrew Gordon Wilson. "GPyTorch: Blackbox 
Matrix-Matrix Gaussian Process Inference with GPU
Acceleration." In Advances in Neural Information Processing
Systems (2018).

**Comment by jacobrgardner** [comment]
The `InvQuadLogDet` Function you mention utilizes this modified CG to compute the GP marginal log likelihood. If you are specifically looking for the lower level function that performs solves and returns tridiagonal matrices, our CG implementation is here: https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/utils/linear_cg.py

There are `lanczos.py` and `stochastic_lq.py` files in the same directory (gpytorch.utils) that deal with getting a log det estimate from the tridiagonal matrices.

Note that if you use the preconditioner described in the paper, your probe vectors will have to be drawn from `N(0, P)`, not `N(0, I)` (see the implementation of InvQuadLogDet).

**Comment by joacorapela** [comment]
Thanks Jake. Very useful hints.
